azerothcore/azerothcore-wotlk
# fix(Core/TheMechanar): Fix Polarity shift mechanic

## Changes Proposed:
-  Codestyle
-  Change the stack amount of the dmg % spell depending on how many players are nearby.
-  Remove the charges once the polarity fades.

## Issues Addressed:

- Closes https://github.com/chromiecraft/chromiecraft/issues/5734

## SOURCE:

https://youtu.be/nupalILx5Zw?t=721

## Tests Performed:

- Tested in-game, further test appreciated.

## How to Test the Changes:


1. Go to The Mechanar and engage Mechano Lord.
2. Wait for Polarity Shift.
3. Test with different amount of players nearby, the stacks should change everytime.
4. End the fight, once the debuff expires, it should remove the stacks.
